### PR TITLE
chore: make sha1 the default algo

### PR DESCRIPTION
**Pull request by kurtextrem**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@fastify/etag",
-  "version": "4.2.0",
+  "version": "5.0.0",
```
Comment (mcollina) [inline package.json:3]: please revert this, I'll bump the major when releasing.

File: fnv1a.js [hunk 2]
```diff
@@ -1,64 +0,0 @@
-'use strict'
```
Comment (mcollina) [inline fnv1a.js:1]: Please retain this file

File: README.md [hunk 3]
```diff
@@ -41,22 +41,14 @@ app.listen(3000)
...
-The fnv1a logic was forked from https://github.com/sindresorhus/fnv1a
-and adapted to support buffers.
-
 ## Benchmarks
 
-* `md5` algorithm: 29679 req/s (median)
-* `sha1` algorithm: 25935 req/s (median)
-* `fnv1a` algorithm: 42943 req/s (median)
-* No ETag generation: 45471 req/s (median)
+Generating an etag will always be slower than not generating an etag and performance depends on the payload size.
```
Comment (mcollina) [inline README.md:50]: Please include a longer explanation, lifting some text from the linked issue.

File: README.md [hunk 4]
```diff
@@ -41,22 +41,14 @@ app.listen(3000)
 
 ## Plugin Options
 
-* `algorithm`: all hashing algorithm that Node.js support, and
-  `'fnv1a'`. Default: `'fnv1a'`.
+* `algorithm`: all hashing algorithms the Node.js [`crypto`](https://nodejs.org/api/crypto.html) module supports. Default: `'sha1'`.
```
Comment (mcollina) [inline README.md:44]: let's tell users to set `fnv1a` in case they have incredibly short payloads.

**Review by mcollina — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (kurtextrem) [reply package.json:3]: oh, didn't know that, will do. If we retain fnv1a it's also not breaking anyway (I guess a minor bump would be fine).

**Comment by kurtextrem** [comment]
Updated according to your review and added recommendations based on our convo in #91

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by kurtextrem** [comment]
had to fix a linter warning, sorry (my git removed a newline from fnv1a.js)

**Review by mcollina — APPROVED** [review]
lgtm

**Review by climba03003 — APPROVED** [review]

**Comment by Uzlopak** [comment]
We could add a validation for the algorithm. So that we can throw on plugin registration and not on the first request.

Like https://github.com/fastify/fastify-cookie/blob/485db2a5676eb7c839fe9687db230777f4077a69/signer.js#L34

**Comment by Uzlopak** [comment]
LOL we have that already.

**Review by Uzlopak — APPROVED** [review]
LGTM

**Pull request merged**